app/forms/openid_connect_authorize_form.rb

```diff
@@ -64,7 +65,11 @@ def ial2_service_provider?
   end
 
   def ial2_requested?
-    ial == 2
+    ial == 2 && !service_provider.liveness_checking_required
+  end
+
+  def ial2_strict_requested?
+    ial == 22 || (ial == 2 && service_provider.liveness_checking_required)
```

Comment:
I honestly think it would be worth it for us to migrate away from this int column into a varchar

Failing that, can we make a constant for IAL2_STRICT_INT or something? Because 22 is not intuitive to me

Reply:
Also I see we have essentially this same ial checking logic everywhere.... maybe it's time for an `IalContext` object?

```ruby
ial_context = IalContext.new(ial: ial, service_provider: service_provider)

ial_context.ial2_requested?
ial_context.ial2_strict_requested?
```
```

Reply:
We could definitely change the column to a varchar in a separate story.  We wouldn't be changing it though we'd be migrating it to a new column so it's not trivial.

Reply:
Yeah after the smoke cleared I noticed it needed to be a little DRYer... not even sure I need most of the checks now that it's in session